src/plugins/expressions/common/execution/execution.ts

```diff
@@ -201,12 +216,40 @@ export class Execution<
     };
 
     this.result = this.input$.pipe(
-      switchMap((input) => this.race(this.invokeChain(this.state.get().ast.chain, input))),
+      switchMap((input) =>
+        this.race(this.invokeChain(this.state.get().ast.chain, input)).pipe(
+          (source) =>
+            new Observable<ExecutionResult<Output>>((subscriber) => {
+              let latest: ExecutionResult<Output> | undefined;
+
+              subscriber.add(
+                source.subscribe({
+                  next: (result) => {
```

Comment:
It seems like sources need to provide their own partial status indicator here, not just automatically setting it to `partial: true`.

Reply:
That's being [overwritten](https://github.com/elastic/kibana/pull/102403/files#diff-47bf3af7472c08ede90f86575ee7db4b8a2c949b1fb61400410ed16a8d819667R234) later on the same frame upon completion. So for the last result that will be `true`. You can also see that in the unit test.